<a href="https://colab.research.google.com/github/Jett987/Discrete_Mathematics/blob/main/McDowellJett_CPSMA_3913_01_Module4_Counting.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Project Part # 4 : CPSMA 3913 Dr. Lynam's Discrete Mathematics Project Part#4
Jett McDowell
(all coding was referenced from Dr. Jacob's github Discrete_Math python repository.)

# Project Part # 4: CPSMA 3913 Dr. Lynam's Discrete Mathematics utilizing [Dr. Jacob's](https://github.com/nurfnick) Discrete Mathematics github repository

## Project Module 4: Counting

### Jett McDowell  206812 

![Lunch.jpg photo courtesy: Microsoft Word picture stock images](https://raw.githubusercontent.com/Jett987/Discrete_Mathematics/main/Lunch%20Picture2.jpg)

(Picture Credit: Microsoft Word Picture Stock Images)



##The answers from section $2$ are answered throughout the document. 

$2a.$ Describe your findings in words.
$2b.$ If you were unable to code something discuss what the difficulty was and why doing it by hand was easier. 

## 1.) Consider a simulation of 10 events with 20 possible outcomes. 

For my simulation I'll just continue thinking of the lunch scenario I created in the previous module, but this time I need to imagine more events. 

I'll begin with a few of the required packages for my project. 

In [30]:
import math
from sympy import *
x = symbols('x')

###Part a.) How many outcomes were possible in your simulation if it is considered a sequence in that particular order? 

If I draw 10 individual lines, and put 20 outcomes in each, I would end up with $20^{10}$ possible outcomes. 

In [31]:
20**10

10240000000000

That is quite a lot of outcomes! 

###Part b.) How many outcomes were possible in your simulation if order did not matter? 

The question here is slightly confusing to me. Initially, I thought of using the equation listed below. 

Since order doesn't matter, $nCk$, or $\frac{n!}{(n-k)!k!}$, seems like the appropriate way to calculate this answer.


In [32]:
math.factorial(20)/(math.factorial(10)*math.factorial(10))

184756.0

After further review, I believe the question is trying to make use of $\binom{n+k-1}{n}$ which yields the following result. 

In [33]:
math.factorial(29)/(math.factorial(20)*math.factorial(9))

10015005.0

###Part c.) How many outcomes were possible if you were not allowed to select an item multiple times? Give two answers, with order mattering and not. 

If I am not allowed to select an item multiple times, and order matters, I will use $nPr$ or $\frac{n!}{(n-k)!}$ to calculate the answer as demonstrated below.

I have 20 choices, I want to pick 10 of them, so I end up with $\frac{20!}{(20-10)!}$. 

In [34]:
math.factorial(20)/(math.factorial(10))

670442572800.0

If order doesn't matter I am going to use $nCk$ to calculate this answer, or $\frac{n!}{(n-k)!k!}$ which gives me $\frac{20!}{(20-10)!10!}$.


In [35]:
math.factorial(20)/(math.factorial(10)*math.factorial(10))

184756.0

###Part d.) What is the coefficient of the 5th order of $(x-3)^9$

If I use the binomial coefficient given in the lecture video, I can apply the following formula to obtain a solution. 

$\binom{n}{i}*x^i*y^{n-i}$

Which would then give me:

$9C5 * x^5 * (-3) ^{(9-5)}$



In [36]:
ninechoosefive = math.factorial(9)/(math.factorial(4)*math.factorial(5))

y=(-3)**4

x=x**5

ninechoosefive

126.0

If I multiply them all together, I get $x^5*10206$. 

In [37]:
ninechoosefive*x*y

10206.0*x**5

This is different than what I thought it would be. I thought I would use $\binom{n}{i}*x^{n-i}*y^{i}$ which would have given me $9C5*x^4*(-3)^5$

In [38]:
ninechoosefive2 = math.factorial(9)/(math.factorial(4)*math.factorial(5))

y2=(-3)**5

x2=x**4

ninechoosefive2

126.0

In [39]:
product = ninechoosefive2*(y2)

product

-30618.0

The result would have been $x^4 * -30618$.

![Lunch2.jpg photo courtesy: Microsoft Word picture stock images](https://raw.githubusercontent.com/Jett987/Discrete_Mathematics/main/Lunch3.jpg)

(Picture Credit: Microsoft Word Picture Stock Images)

##$2c.$ Why are all these outcomes different?

The outcomes were all different, because each answer is reliant on a specific set of rules. If order matters, you must use permutation, if order doesn't matter, you should use combination, and if you are trying to evaluate the amount of possible outcomes you use the amount of outcomes possible and how many events they can be applied towards. 